src/better_deferred.js

```diff
@@ -73,48 +59,32 @@ const lozadObserver = () => {
             log(`Script in lozad will be loading now`, 1);
             let queue = fetchBetterDeferred(mutation.target);
             loadScriptsFromQueue(queue);
-            
+
             // Stop observing
-            observer.disconnect();  
+            observer.disconnect();
         }
     };
 
     // Create an observer instance linked to the callback function
     const observer = new MutationObserver(callback);
-    
+
     lozads.forEach((element) => {
         // Start observing the target node for configured mutations
         observer.observe(element, config);
     });
 }
 
-const loadScriptsFromQueue = (queue) => {
-    for(const elem of queue) {
-        let inline = elem.type == INLINE;
-        log(`Processing lozad script - inline? ${inline}`);
-        if (!inline) {
-            addScriptObjects(elem);
-        } else {
-            addInlineScriptObjects(elem);
-        }
-    }
-}
+const loadScriptsFromQueue = (queue) => queue.forEach((elem) => addScriptObjects(elem));
 
 const fetchBetterDeferred = (parent = document) => {
-    let queue = [];
-    let externalScripts = parent.querySelectorAll("script[type=" + DEFERRED + "]");
-    externalScripts.forEach((element) => {
-        queue.push(element);
-    });
-
-    let inlineScripts = parent.querySelectorAll("script[type=" + INLINE + "]");
-    inlineScripts.forEach((element) => {
-        queue.push(element);
-    });
+    const queue = Array.from(parent.querySelectorAll(`script[type^="${DEFERRED}"]`));
 
-    log(`Found ${externalScripts.length} ext scripts, ${inlineScripts.length} inline scripts. Queue ${queue.length}`, 1);
-    return queue;
-};
+    // Since we want to preserve the load order and load external scripts first and then inline scripts.
+    return [
+        ...queue.filter((elem) => elem.innerHTML === ''),
+        ...queue.filter((elem) => elem.innerHTML !== ''),
```

Comment:
Example: https://playcode.io/1052168

since content of script tag is only evaluated if `src` is missing.

```suggestion
        ...queue.filter(({src}) => src !== ''),
        ...queue.filter(({src}) => src === ''),
```